icu4c/source/i18n/calendar.cpp

```diff
@@ -3948,6 +3948,29 @@ Calendar::setWeekData(const Locale& desiredLocale, const char *type, UErrorCode&
         } else {
             status = U_INVALID_FORMAT_ERROR;
         }
+
+        // Check if the locale has a "fw" u extension and we honor it if present.
+        // And we don't change the overal status, as the presence / lack of "fw" is not an error.
+        UErrorCode fwStatus = U_ZERO_ERROR;
+        char fwExt[ULOC_FULLNAME_CAPACITY] = "";
+        desiredLocale.getKeywordValue("fw", fwExt, ULOC_FULLNAME_CAPACITY, fwStatus);
+        if (!U_FAILURE(fwStatus)) {
```

Comment:
how about U_SUCCESS() instead of !U_FAILURE() ?

Reply:
Done.

---

I know there are (used to be?) warnings states.

So in my mind:
```
if (U_SUCCESS) { ... } => SUCCESS
if (!U_FAILURE) { ... } => SUCCESS || WARNING
```
There are other instances of code doing `if (!U_FAILURE) { ... }`

But it looks like the warning levels were deprecated (protected with `#ifdef`)` and now `U_SUCCESS === ! U_FAILURE`